### ISSUE TITLE
Custom log providers

### ISSUE THREAD

**Issue by scheb** [scheb/tombstone#6]
# Idea

Add configuration option to retrieve log data from a custom log provider. This would give more flexibility to the user how logs are retrieved.

Right now the only way to pass log data from the logger to the analyzer is the "analyzer log format", which is files written to disk. If you want to run the analyzer, either you have to do it on the machine where the log files are located or you need to collect all the log files from those machines. So maybe you want to store logs differently, e.g. write them to a database or push them into some logging infrastructure and then retrieve the data from there.

**If this idea would be useful to you, give it a +1**

**Comment by scheb** [comment]
Feature released with v1.2.0

**Comment by 0nedark** [comment]
@scheb I've tried to use this the other day and I found an issue that makes this unusable. The custom analyzer is initialised through the constructor `$logReader = $reflectionClass->newInstance();` instead of static `create` function that is used to initialise built in `AnalyzerLogProvider`. The issue is that this way the custom analyzer does not have access to the output writer or the config, making it, I think, impossible to create any reports. I suggest the following change to resolve this issue.

```
    private function createLogCollector(array $config, VampireIndex $vampireIndex): LogCollector
    {
        $logProviders = [];
        if (isset($config['logs']['directory'])) {
            $logProviders[] = AnalyzerLogProvider::create($config, $this->output);
        }
        if (isset($config['logs']['custom'])) {
            if (isset($config['logs']['custom']['file'])) {
                /** @psalm-suppress UnresolvableInclude */
                require_once $config['logs']['custom']['file'];
            }

            $className = $config['logs']['custom']['class'];
            $reflectionClass = new \ReflectionClass($className);
            if (!$reflectionClass->implementsInterface(LogProviderInterface::class)) {
                throw new \Exception(sprintf('Class %s must implement %s', $className, LogProviderInterface::class));
            }

            /** @var LogProviderInterface $logReader */
            $logReader = $className::create($config, $this->output);

            $logProviders[] = $logReader;
        }

        return new LogCollector($logProviders, $vampireIndex);
    }
```

**Comment by scheb** [comment]
My bad, I reused some older code and didn't realize the class was instantiated differently. Fix released as v1.2.1.
